### PR TITLE
Declare plugins using a configmap value instead of CRD

### PR DESCRIPTION
**Pull request by alexkursell**
This is an alternate, much simpler PR that reads a JSON array of plugin definitions from a new configmap value, `plugins`.

```yaml
plugins: '[{"archive":"foo.tar.gz", "sha256sum":"aaaaa", "name":"hello_world"}]'
```

This may or may not be the way we want to go, but it certainly is a lot simpler than adding a full CRD for it.

edit: This works, with a few caveats:

- The code still needs unit tests, e2e tests, general cleanup
- The subfolders of a plugin archive to be loaded need to be specified in the configmap as well. This should be autodetected from the folder structure.
- Removed plugins are never deleted, their code remains in the container forever

### CODE REVIEW

File: internal/ingress/controller/nginx.go [hunk 1]
```diff
@@ -814,6 +814,9 @@ func (n *NGINXController) IsDynamicConfigurationEnough(pcfg *ingress.Configurati
 	copyOfRunningConfig.Backends = []*ingress.Backend{}
 	copyOfPcfg.Backends = []*ingress.Backend{}
 
+	copyOfRunningConfig.Plugins = []ingress.Plugin{}
+	copyOfPcfg.Plugins = []ingress.Plugin{}
```
Comment (ElvinEfendi) [inline internal/ingress/controller/nginx.go]: let's not skip reload

Comment (alexkursell) [reply internal/ingress/controller/nginx.go]: ok

File: internal/ingress/controller/config/config.go [hunk 2]
```diff
@@ -276,6 +276,9 @@ type Configuration struct {
 	// http://nginx.org/en/docs/http/ngx_http_core_module.html#server_names_hash_bucket_size
 	ServerNameHashBucketSize int `json:"server-name-hash-bucket-size,omitempty"`
 
+	// A JSON array of lua plugin definitions to be executed for each request
+	Plugins []ingress.Plugin `json:"plugins,omitempty"`
```
Comment (ElvinEfendi) [inline internal/ingress/controller/config/config.go]: Why in the middle of others but not in the end?

File: internal/ingress/controller/controller.go [hunk 3]
```diff
@@ -108,6 +110,27 @@ func (n NGINXController) GetPublishService() *apiv1.Service {
 	return s
 }
 
+// FetchChangedPlugins tries to download every plugin that didn't previously exist
```
Comment (ElvinEfendi) [inline internal/ingress/controller/controller.go]: nitpick: ... didn't previously exist or got changed

File: internal/ingress/controller/plugin/storage_gcs.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+var (
+	// ErrInvalidGCSURL is returned when the gs://... url is invalid
+	ErrInvalidGCSURL = errors.New("invalid gcs URL")
+)
+
+type gcsProvider struct {
+}
+
+// Parse returns the bucket and filename from a gcs URL.
+func (p *gcsProvider) Parse(archiveURL string) (bucket, filename string, err error) {
```
Comment (ElvinEfendi) [inline internal/ingress/controller/plugin/storage_gcs.go]: it is not a good practice to use implicit return

**Pull request merged**